Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -411,7 +411,16 @@ class BrowserViewController: UIViewController {
             self.view.alpha = (profile.prefs.intForKey(IntroViewControllerSeenProfileKey) != nil) ? 1.0 : 0.0
         }
 
-        if activeCrashReporter?.previouslyCrashed ?? false {
+        // Only ask to restore tabs from a crash if we had non-home tabs or tabs with some kind of history in them
+        guard let tabsToRestore = tabManager.tabsToRestore() else { return }
```

Comment:
Ah - yes I did notice this. I don't think we need to call updateTabCount since the number of tabs is zero so it essentially is a no-op.